airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -506,15 +509,17 @@ def get_latest_serialized_dags(
             .group_by(cls.dag_id)
             .subquery()
         )
-        latest_serdags = session.scalars(
-            select(cls)
-            .join(
-                latest_serdag_subquery,
-                cls.created_at == latest_serdag_subquery.c.created_at,
-            )
-            .where(cls.dag_id.in_(dag_ids))
-        ).all()
-        return latest_serdags or []
+        latest_serdags: list[SerializedDagModel] = list(
+            session.scalars(
+                select(cls)
+                .join(
+                    latest_serdag_subquery,
+                    cls.created_at == latest_serdag_subquery.c.created_at,
+                )
+                .where(cls.dag_id.in_(dag_ids))
+            ).all()
+        )
```

Comment:
Please do not do this. Change the return annotation to Sequence[SerializedDagModel] instead.